### PR TITLE
feat: install-target parser for wengine install shorthands

### PR DESCRIPTION
**Pull request by xujustinj**
## Summary

PR 2 of the node-distribution series (`docs/plans/node-distribution.md`). Parses the operator-facing `wengine install <target>` strings into a structured `InstallTarget` and turns each variant into an argv tail for `uv add`.

- `parse_install_target(s)` → `PyPITarget | GitTarget | PathTarget | ForgeTarget` (all `BaseInstallTarget` subclasses). Pure, no I/O.
  - PyPI: bare names, version specifiers, extras (`acme-scrapers[screenshot]==1.4.0`) — passed through to `uv add` as-is.
  - Git: `git+https://...@ref[#subdirectory=...]` — passed through.
  - Path: `./path`, `../path`, `/abs/path`, or `path:<anything>` — `uv add --editable`.
  - Forge: `github:owner/repo[@ref][#subdirectory=...]`, `gitlab:...`, and bare `owner/repo` (defaults to GitHub).
- Every variant exposes `uv_add_args` as a `cached_property`. For PyPI/Git/Path it's a trivial accessor; for `ForgeTarget` it resolves the ref over HTTPS to a commit SHA, then returns the pinned tarball URL. uv reads the archive metadata itself, so no separate distname lookup is needed.
- `resolve_forge_ref(target)` is also exposed directly. It calls a module-private `_http_get`; tests monkeypatch that name rather than passing an injected getter.

Only the public `github.com` / `gitlab.com` hosts are supported; self-hosted GitLab and GitHub Enterprise users should reach for `git+https://` instead.

## What this PR does *not* do

Nothing user-visible yet — no CLI command wires this up. That lands in PR 3 (`uv` integration / `wengine install`). This PR is structural pre-work, kept standalone so its parsing edge cases can be reviewed and tested in isolation.

## Test plan

- [x] `uv run pytest tests/test_install_target.py` — 27 tests covering every shorthand, malformed input, default-branch lookup, URL-encoded refs, both forges, and `ForgeTarget.uv_add_args` end-to-end with `_http_get` monkeypatched
- [x] Full suite passes
- [x] `ruff check` / `ruff format` / `pyright` clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/workflow_engine/cli/install_target.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+from workflow_engine.utils.model import ImmutableBaseModel
+
+ForgeName = Literal["github", "gitlab"]
+
+_FORGE_TARBALL_URL_TEMPLATES: Mapping[ForgeName, str] = {
+    "github": "https://codeload.github.com/{owner}/{repo}/tar.gz/{sha}",
+    # GitLab's archive endpoint: the trailing filename is cosmetic but required
+    # by the route, and conventionally `<repo>-<sha>.tar.gz`.
+    "gitlab": "https://gitlab.com/{owner}/{repo}/-/archive/{sha}/{repo}-{sha}.tar.gz",
+}
```
[the remark this answers is not in the document]
Comment (xujustinj) [reply src/workflow_engine/cli/install_target.py:41]: well we already have `git+` for private remotes, so this really is a non-issue
